File: mar_28/R/page1.R
printAccuracy = function(cm) {
  correct.predictions = sum(diag(cm))
  total.predictions = sum(cm)
  accuracy = (correct.predictions / total.predictions) * 100
  print(accuracy)
}

# step 1: load the data
df = read.csv('/Volumes/Data/Sunbeam/2019/Feb/DBDA/stats_R/dataset/data\ 1/Social_Network_Ads.csv')

# step 2: clean the data
df = df[3:5]

library(caTools)
results = sample.split(df$Purchased, SplitRatio = 0.8)
df.train = df[results == TRUE, ]
df.test = df[results == FALSE, ]

# step 3: prepare the model/formula/equation
library(rpart)

# greater the minsplit greater the accuracy
classifier = rpart(Purchased ~ ., data = df.train, control = rpart.control(minsplit = 15))

plot(classifier)
text(classifier)

# step 4: predict the value(s)
predictions = predict(classifier, newdata = df.test)
predictions = ifelse(predictions >= 0.5, 1, 0)

# step 5: test the result
cm = table(df.test$Purchased, predictions)
printAccuracy(cm)

# step 6: visualize the result

File: mar_28/python/page1.py
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

def printAccuracy(cm):
    total = sum(sum(cm))
    accurate = sum(np.diagonal(cm))
    accuracy = (accurate / total) * 100
    print('accuracy: {}%'.format(accuracy))

# step 1: load the data
df = pd.read_csv('/Volumes/Data/Sunbeam/2019/Feb/DBDA/stats_R/dataset/data 1/Social_Network_Ads.csv')
x = df.iloc[:, 2:4].values
y = df.iloc[:, 4].values

# step 2: clean the data
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, random_state=3434, test_size=0.2)

# step 3: prepare the model/formula/equation
from sklearn.tree import DecisionTreeClassifier
classifier = DecisionTreeClassifier(criterion='gini')
classifier = classifier.fit(x_train, y_train)

# from sklearn.tree import DecisionTreeRegressor
# classifier = DecisionTreeRegressor()
# classifier = classifier.fit(x_train, y_train)

# step 4: predict the value(s)
predictions = classifier.predict(x_test)

# step 5: test the result
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test, predictions)
printAccuracy(cm)

# step 6: visualize the result